aloha-cs-avro/src/main/scala/com/eharmony/aloha/semantics/compiled/plugin/avro/AvroCodeGenerators.scala

```diff
@@ -0,0 +1,109 @@
+package com.eharmony.aloha.semantics.compiled.plugin.avro
+
+import com.eharmony.aloha.reflect.RefInfoOps
+import com.eharmony.aloha.semantics.compiled.plugin.schemabased.accessor._
+import com.eharmony.aloha.semantics.compiled.plugin.schemabased.codegen.{AccessorCodeGenerator, CodeGenerators}
+import com.eharmony.aloha.semantics.compiled.plugin.schemabased.schema._
+
+
+/**
+  * '''Note''': We use the named extractors rather than intege-based extractors to extract fields because
+  * of the note in the [[https://avro.apache.org/docs/current/spec.html#Schema+Resolution Schema Resolution]]
+  * section of the Avro specification which states:
+  *
+  * "''the ordering of fields may be different: fields are matched by name.''"
+  *
+  * This is probably slower but should be safer.
+  */
+private[avro] object AvroCodeGenerators extends CodeGenerators {
+  def castCode(fd: FieldDesc): String = {
+    def fieldTypeString(fd: FieldDesc): String = {
+      val typeStr = fd match {
+        case f: RecordField => RefInfoOps.toString(f.refInfo)
+        case f: StringField => "org.apache.avro.util.Utf8"  // Special, b/c it's mutable (unlike String)!!!
+        case f: IntField => "Int"
+        case f: LongField => "Long"
+        case f: FloatField => "Float"
+        case f: DoubleField => "Double"
+        case f: BooleanField => "Boolean"
+        case f: ListField => "java.util.List[" + fieldTypeString(f.elementType) + "]" // OK, Avro implements List.
+        case f: EnumField => throw new IllegalStateException("enum not supported")
```

Comment:
Aren't there other types that Avro supports (such as Map)?  While I don't think we need to support them now we should at least mention them all here so we can throw an appropriate error.